rmw_zenoh_cpp/src/detail/graph_cache.cpp

```diff
@@ -84,12 +91,31 @@ void GraphCache::parse_put(const std::string & keyexpr)
       }
 
       const liveliness::TopicInfo topic_info = entity.topic_info().value();
-      GraphNode::TopicMap & topic_map = entity.type() ==
-        EntityType::Publisher ? graph_node.pubs_ : graph_node.subs_;
-      const std::string entity_desc = entity.type() ==
-        EntityType::Publisher ? "publisher" : "subscription";
-      const std::size_t pub_count = entity.type() == EntityType::Publisher ? 1 : 0;
+      std::string entity_desc = "";
+      GraphNode::TopicMap & topic_map =
+        [&]() -> GraphNode::TopicMap &
+        {
+          if (entity.type() == EntityType::Publisher) {
+            entity_desc = "publisher";
+            return graph_node.pubs_;
+          } else if (entity.type() == EntityType::Subscription) {
+            entity_desc = "subscription";
+            return graph_node.subs_;
+          } else if (entity.type() == EntityType::Service) {
+            entity_desc = "service";
+            return graph_node.services_;
+
+          } else {
+            entity_desc = "client";
+            return graph_node.clients_;
+          }
+        }();
+      // For the sake of reusing data structures and lookup functions, we treat publishers and clients are equivalent.
+      // Similarly, subscriptions and services are equivalent.
```

Comment:
I'm somewhat suspicious of this assertion, as they aren't the same thing.  I'm worried that this detail is going to leak out through the RMW APIs, though I haven't tested it directly yet to tell for certain.

Reply:
The alternative is to have separate data structures  and functions for clients and services. It will create unnecessary code duplications but improves readability and maintenance.

Reply:
My main concern is that this is going to leak through to things like "publisher count" and "client count".  But let's stick with what you have for now, and we can update it later if that is a problem.